cylc/flow/config_schema.py

```diff
@@ -0,0 +1,673 @@
+    u.obsolete("7.2.2", ["cylc", "dummy mode"])
+    u.obsolete("7.2.2", ["cylc", "simulation mode"])
+    u.obsolete("7.2.2", ["runtime", "__MANY__", "dummy mode"])
+    u.obsolete("7.2.2", ["runtime", "__MANY__", "simulation mode"])
+    u.obsolete("7.6.0", ["runtime", "__MANY__", "enable resurrection"])
+    u.obsolete(
+        "7.8.0", ["runtime", "__MANY__", "suite state polling", "template"]
+    )
+    u.obsolete("7.8.0", ["suite logging", "roll over at start-up"])
+    u.obsolete("7.8.1", ["cylc", "events", "reset timer"])
+    u.obsolete("7.8.1", ["cylc", "events", "reset inactivity timer"])
+    u.obsolete("7.8.1", ["runtime", "__MANY__", "events", "reset timer"])
+    u.obsolete("8.0.0", ["cylc", "log resolved dependencies"])
+    u.obsolete("8.0.0", ["cylc", "reference test", "allow task failures"])
+    u.obsolete("8.0.0", ["cylc", "reference test", "live mode suite timeout"])
+    u.obsolete("8.0.0", ["cylc", "reference test", "dummy mode suite timeout"])
+    u.obsolete(
+        "8.0.0", ["cylc", "reference test", "dummy-local mode suite timeout"]
+    )
+    u.obsolete(
+        "8.0.0", ["cylc", "reference test", "simulation mode suite timeout"]
+    )
+    u.obsolete("8.0.0", ["cylc", "reference test", "required run mode"])
+    u.obsolete(
+        "8.0.0", ["cylc", "reference test", "suite shutdown event handler"]
+    )
+    u.deprecate(
+        "8.0.0",
+        ["cylc", "abort if any task fails"],
+        ["cylc", "events", "abort if any task fails"],
+    )
+    u.obsolete("8.0.0", ["runtime", "__MANY__", "job", "shell"])
+    u.upgrade()
+
+    # Upgrader cannot do this type of move.
+    try:
+        keys = set()
+        cfg["scheduling"].setdefault("graph", {})
+        cfg["scheduling"]["graph"].update(
+            cfg["scheduling"].pop("dependencies")
+        )
+        graphdict = cfg["scheduling"]["graph"]
+        for key, value in graphdict.copy().items():
+            if isinstance(value, dict) and "graph" in value:
+                graphdict[key] = value["graph"]
+                keys.add(key)
+        if keys:
+            LOG.warning(
+                "deprecated graph items were automatically upgraded in '%s':",
+                descr,
+            )
+            LOG.warning(
+                " * (8.0.0) %s -> %s - for X in:\n%s",
+                u.show_keys(["scheduling", "dependencies", "X", "graph"]),
+                u.show_keys(["scheduling", "graph", "X"]),
+                "\n".join(sorted(keys)),
+            )
+    except KeyError:
+        pass
+
+
+class RawSuiteConfig(ParsecConfig):
+    """Raw suite configuration."""
+
+    def __init__(self, fpath, output_fname, tvars):
+        """Return the default instance."""
+        ParsecConfig.__init__(
+            self, SPEC, upg, output_fname, tvars, cylc_config_validate
+        )
+        self.loadcfg(fpath, "suite definition")
+
+
+class GlobalConfig(ParsecConfig):
+    """
```

Comment:
Strategy - @oliver-sanders  is it worth doing the work to combine this with plain old config.py next? This feels like it might be a sensible strategy....

Reply:
If by `config.py` you mean `RawSuiteConfig` then maybe yes. We don't need two different classes to represent the same config, the only difference now is the loading logic.